cmake/android/android_gradle_projects.cmake

```diff
@@ -1,8 +1,8 @@
 # https://developer.android.com/studio/releases/gradle-plugin
-set(ANDROID_GRADLE_PLUGIN_VERSION "3.2.1" CACHE STRING "Android Gradle Plugin version")
+set(ANDROID_GRADLE_PLUGIN_VERSION "4.1.2" CACHE STRING "Android Gradle Plugin version")
 message(STATUS "Android Gradle Plugin version: ${ANDROID_GRADLE_PLUGIN_VERSION}")
 
-set(GRADLE_VERSION "5.6.4" CACHE STRING "Gradle version")
+set(GRADLE_VERSION "6.8.2" CACHE STRING "Gradle version")
```

Comment:
see https://github.com/opencv/opencv/pull/19501#issuecomment-777490097

Reply:
I don't understand. Sorry to say this, but your pipeline for Android is over engineered and comes with too much complexity. 
What's about Github actions ?

Reply:
Under "too much complexity" we have maintained backward compatibility.
Not sure how "GitHub actions" may help here (we still need to upgrade dependencies installation part in these scripts too).

These CMake values should be specified externally through config files. Keep "backward compatible" default values in CMake file untouched.

Reply:
I mean with Github action there is no need of license-accept-something, simply do this (It's valid for all api)

https://github.com/hannesa2/LiveEdgeDetection/blob/master/.github/workflows/Android-CI-Espresso.yml#L27-L28

To run tests is similar easy 
https://github.com/hannesa2/LiveEdgeDetection/blob/master/.github/workflows/Android-CI-Espresso.yml#L27-L28

But it's just for Android

Reply:
GitHub actions builds environments are constructed on the fly from the scratch (with some help of caching).

OpenCV builds configurations usually have locked dependencies (fixed predefined build environment is preferred).
Build environments are updated separately (through build infrastructure update process).
Next upgrade for Android dependencies is expected with 5.0 major release (see the "next" branch). Feel free to open issue with proposed Android dependencies upgrade or add comment here: #18995